### PR TITLE
Simplify image builds by using QEMU to do multi-arch builds

### PR DESCRIPTION
**Pull request by chirino**
but since they are slow, only do the multi-arch builds for releases.

Also support passing cargo build args to the Docker image builder, allowing us to enable all features.

### CODE REVIEW

File: .github/workflows/build-image.yaml [hunk 1]
```diff
@@ -6,8 +6,6 @@ on:
   push:
     branches:
       - main
-    tags:
-      - "*"
```
Comment (alexsnaps) [inline .github/workflows/build-image.yaml:10]: I think we need this, as that's how we release, no? @didierofrivia 
tags could be only `v[0-9]+\.[0-9]+\.[0-9]+(-.*)?` or something tho

File: .github/workflows/build-multi-arch-image.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+name: Build Image
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - "*"
```
Comment (chirino) [inline .github/workflows/build-multi-arch-image.yaml:8]: Added another workflow for releases, in this case we build both archs.

**Comment by eguzki** [comment]
If I understand this correctly, image is not being built on PR's. Is this something we want? Building an image is kind of a test as well that either passes or fails.

**Comment by eguzki** [comment]
> If I understand this correctly, image is not being built on PR's. Is this something we want? Building an image is kind of a test as well that either passes or fails.

I realize that the image building was removed from PR's a while ago.

**Comment by didierofrivia** [comment]
I'd say that going back to quemu for the default building is not ideal as the default build job, I'd rather add it as a different `dispatch` workflow and/or only for releases

**Comment by chirino** [comment]
ok.  closing PR, maybe revisit later.

**Pull request closed without merging**